### ISSUE TITLE
When using copy_dir, I think we should make sure that broken symlinks are copied as-is

### ISSUE THREAD

**Issue by wolfv** [prefix-dev/rattler-build#315]
We had a report on Discord that rattler-build was erroring on a broken symlink from the `bun` git repository. I think it might be that we use the `copy_dir` function and it errors when it encounters a broken symlink. Instead, I think we should copy the broken symlink as it is.

**Comment by codegod100** [comment]
Ran into this again

```
Copied /home/vera/conda-builds/gleam-conda/output/bld/rattler-build_gleam_1700524614/build_env/x86_64-conda-linux-gnu/lib/libgcc_s.so to /tmp/gleamIuCZAJ/info/recipe/output/bld/rattler-build_gleam_1700524614/build_env/x86_64-conda-linux-gnu/lib/libgcc_s.so
Error:   × FileSystem error: 'Path "/home/vera/conda-builds/gleam-conda/output/bld/rattler-build_gleam_1700524614/build_env/x86_64-conda-linux-gnu/lib/libgfortran.so" does
  │ not exist or you don't have access!'
 ```

```
vera@debian-4gb-hil-1:~/conda-builds/gleam-conda$ ls -l /home/vera/conda-builds/gleam-conda/output/bld/rattler-build_gleam_1700524614/build_env/x86_64-conda-linux-gnu/lib/libgfortran.so
lrwxrwxrwx 1 vera vera 24 Nov 20 23:56 /home/vera/conda-builds/gleam-conda/output/bld/rattler-build_gleam_1700524614/build_env/x86_64-conda-linux-gnu/lib/libgfortran.so -> ../../lib/libgfortran.so
```

**Comment by wolfv** [comment]
fixed in #341
